### PR TITLE
[SYCLomatic] Migration of cudaGraphInstantiate,cudaGraphLaunch,cudaGraphExecDestroy

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

**Review by intwanghao — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -15085,3 +15085,22 @@ void AssertRule::runRule(const MatchFinder::MatchResult &Result) {
...
+    return hasAnyName("cudaGraphInstantiate", "cudaGraphLaunch",
+                      "cudaGraphExecDestroy");
+  };
+  MF.addMatcher(
+      callExpr(callee(functionDecl(functionName()))).bind("FunctionCall"),
+      this);
+}
+
+void GraphRule::runRule(const MatchFinder::MatchResult &Result) {
+  const CallExpr *CE = getNodeAsType<CallExpr>(Result, "FunctionCall");
```
Comment (tangjj11) [inline clang/lib/DPCT/ASTTraversal.cpp:15100]: Should add `if (CE != nullptr)`?

File: clang/lib/DPCT/APINamesGraph.inc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+CONDITIONAL_FACTORY_ENTRY(
+    UseExtGraph,
+    ASSIGNABLE_FACTORY(ASSIGN_FACTORY_ENTRY(
+        "cudaGraphInstantiate", DEREF(0),
+        NEW(MapNames::getClNamespace() +
+                "ext::oneapi::experimental::command_graph" + "<" +
```
Comment (tangjj11) [inline clang/lib/DPCT/APINamesGraph.inc]:
```suggestion
                "ext::oneapi::experimental::command_graph<" +
```

**Review by tangjj11 — APPROVED** [review]

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 1 above)
Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/ASTTraversal.cpp:15100]: Done.

File: clang/lib/DPCT/APINamesGraph.inc (hunk 2 above)
Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/APINamesGraph.inc]: Done.

**Review by ziranzha — APPROVED** [review]

**Pull request merged**